### PR TITLE
carli/1947_mirror_cursor_when_spatially_matched

### PR DESCRIPTION
**Pull request by crocka**
**Description**
This is an enhancement functionality. The cursor is mirrored on all other spatially matched images to allow users to see the current corresponding cursor position on other images. An boolean observable is added to the AppStore called cursorMirror to keep track of the functionality status, and a button is added to the cursorInfo widget to toggle on/off. Once the functionality is turned on, a cross cursor appears on the reference image and it follows the user's cursor. When the user turn on the spatial matching to the reference image, a cross cursor appears on the corresponding image. Users can hover on any of the images that are spatially matched to each other to see the corresponding cursor location on all other spatially matched frames.

**Checklist**
- [x] changelog updated / ~~no changelog update needed~~
- [x] e2e test passing / ~~added corresponding fix~~
- [x] protobuf updated to the latest dev commit / ~no protobuf update needed~

### CODE REVIEW

**Comment by kswang1029** [comment]
few initial comments:
1. we cannot use space as the hotkey as it can trigger image matching if we just click the matching button in the image list then press space key, for example. Maybe we can consider the G key as it is next to the F key, although G means nothing here (or "group"). Open to other ideas. 

https://user-images.githubusercontent.com/20819712/188296401-99f67169-6ecb-492b-88d5-7e273aaba315.mov

2. if we enable cursor mirroring _and_ trigger cursor freezing with the F key, we will see a cross marker at an unexpected location on the unmatched images. This is nothing to do with this PR and this behavior exists in the dev. I suggest we use a different color for the matched _and_ mirrored cross just to differentiate this case.

Screen Shot 2022-09-04 at 11 46 22

**Comment by kswang1029** [comment]
@crocka feature-wise this is good for a PR. 👍

File: src/components/ImageView/RegionView/CursorRegionComponent.tsx [hunk 1]
```diff
@@ -15,13 +15,16 @@ interface CursorRegionComponentProps {
 @observer
 export class CursorRegionComponent extends React.Component<CursorRegionComponentProps> {
     render() {
+        const appStore = AppStore.Instance;
         const frame = this.props.frame;
         const posImageSpace = frame?.cursorInfo?.posImageSpace;
+        //check the current frame is the spatialReg
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/RegionView/CursorRegionComponent.tsx]: typo for spatialRef?

File: src/components/ImageView/RegionView/CursorRegionComponent.tsx [hunk 2]
```diff
@@ -15,13 +15,16 @@ interface CursorRegionComponentProps {
...
+        const isSpatialMatchingOn = appStore.frames.some(thisFrame => thisFrame.spatialReference === frame);
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/RegionView/CursorRegionComponent.tsx]: Perhaps we can reuse `frame.secondarySpatialImages?.length`

**Review by YuHsuan-Hwang — COMMENTED** [review]
looks good, just minor comments

File: src/components/ImageView/RegionView/CursorRegionComponent.tsx (hunk 1 above)
Comment (crocka) [reply src/components/ImageView/RegionView/CursorRegionComponent.tsx]: Yes, it is a typo. It is fixed now.

File: src/components/ImageView/RegionView/CursorRegionComponent.tsx (hunk 2 above)
Comment (crocka) [reply src/components/ImageView/RegionView/CursorRegionComponent.tsx]: Yes, it works the same way. It is updated now.

**Review by kswang1029 — APPROVED** [review]
feature-wise this is good. no regression is observed via e2e tests.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
@crocka Please merge dev and update changelog.

**Pull request merged**